x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/stream_enrichment_state_machine/stream_enrichment_state_machine.ts

```diff
@@ -0,0 +1,355 @@
+  id: 'enrichStream',
+  context: ({ input }) => ({
+    definition: input.definition,
+    initialProcessorsRefs: [],
+    processorsRefs: [],
+  }),
+  initial: 'initializing',
+  states: {
+    initializing: {
+      always: [
+        {
+          target: 'resolvedRootStream',
+          guard: 'isRootStream',
+        },
+        { target: 'ready' },
+      ],
+    },
+    ready: {
+      id: 'ready',
+      type: 'parallel',
+      entry: enqueueActions(({ check, enqueue }) => {
+        enqueue({ type: 'stopProcessors' });
+        enqueue({
+          type: 'setupProcessors',
+          params: ({ context }) => ({ definition: context.definition }),
+        });
+        // Setup fields for wired stream only
+        if (check('isWiredStream')) {
+          enqueue({
+            type: 'setupFields',
+            params: ({ context }) => ({
+              definition: context.definition as WiredStreamGetResponse,
+            }),
+          });
+        }
+      }),
+      on: {
+        'stream.received': {
+          target: '#ready',
+          actions: [{ type: 'storeDefinition', params: ({ event }) => event }],
+          reenter: true,
+        },
+      },
+      states: {
+        stream: {
+          initial: 'idle',
+          states: {
+            idle: {
+              on: {
+                'stream.reset': {
```

Comment:
When resetting, we need to update the simulation state machine - right now it stays in its current state.

Reply:
++, it's part of the changes I'm doing to better assert the simulation